meson.build

```diff
@@ -119,7 +119,6 @@ mandir = join_paths(prefix, get_option('mandir'))
 gio = dependency('gio-2.0', version : '>= 2.45.8')
 giounix = dependency('gio-unix-2.0', version : '>= 2.45.8', required: false)
 lzma = dependency('liblzma')
-zstd = dependency('libzstd')
```

Comment:
You could simplify the code by just using something like this:
``` meson
zstd = dependency('libzstd')
if get_option('zstd')
    add_project_arguments('-DENABLE_ZSTD', language: 'c')
    if not zstd.found() 
        error('libzstd is missing, please install it to continue!')
    endif
endif
``` 
That avoids having the extra `libxmlb_deps` variable.
By adding a `HAVE_ZSTD` define directly to the project's `configuration_data` above, you can simplify the code even more and have an even smaller Meson definition (with only `extra_sources remaining).

Reply:
I think you could use a meson `feature` too.

Reply:
https://github.com/hughsie/libxmlb/pull/134/commits/f72a109986f0d2ad6dedf4239a7489272bd24b21